Update with new functionalities (settings per device, testmail, help page, sql data export)

avollkopf opened the pull request:
Some functionalities are added such as 
- settings per device functionality: some settings can be separately configured for devices
- testmail button to test email settings (PHPMailer had to be added but not all files are required that are currently in the repo)
- export of diagrams as pdf or image
- sql export of data table (as starting point -> re-import is planned for later. This allows for backup of data)
- Help page (currently only in German). Need to think about on how to allow for multi language support.
- CSS file usage for layout

avollkopf committed:
Summary on Index page and Individual Settings per device

avollkopf committed:
Update Settings.sql

avollkopf committed:
Update with Help, testmail and data export

Data table can be exported es sql file
PHP Mailer is used to send test email via php
Help (currently only in German) is added

avollkopf committed:
Corrected error in common_db_query.php

Additional funtion added to this file for future usage

Commit:
Support for forwarding to InfluxDB and some minor cosmetic changes

DottoreTozzi commented:
There are still some conflicts with these...
I'll do that once I find the time, but that could take a couple of weeks, unfortunately, as I'm insanely busy at the moment.

avollkopf commented:
> 
> 
> There are still some conflicts with these...
> I'll do that once I find the time, but that could take a couple of weeks, unfortunately, as I'm insanely busy at the moment.

No worries, I was also not woerking on my repo for the past couple of weeks and got a pull request for my current repo that I need to inlcude. In addition to that I started now working on the functionality to export and import the data table from the web interface. Export is working. Import may need a weeks or so. Next step is the export of the settings table with the goal to allow an import. Not only for backup but also for re-importing it in case of new versions w/o loosing the existing seetings. Export does not eem to be complex but I need to think about the re-import in more detail.

Maybe I close the pull request and create a new one once these things are completed.

avollkopf committed:
Merge pull request #3 from simonwes/master

Support for forwarding to InfluxDB and some minor cosmetic changes

avollkopf committed:
Import Export of database and settings

-Upgrade of settings table possible in case of new settings
- Upgrade of strings table possible
- Export / Import of database
- Export / Import of Settings possible

avollkopf commented:
I have uploaded a new version to my repo. Hence, I close this pull request.
